src/parser/__tests__/allowed-syntax.ts

```diff
@@ -315,16 +315,44 @@ test.each([
     export default x;
     x;
     `
+  ],
+
+  [
+    Chapter.LIBRARY_PARSER,
+    `
+    function square(x) {
+      return x * x;
+    }
+    export { square as default };
+    `
+  ],
+
+  [
+    Chapter.LIBRARY_PARSER,
+    `
+    import { default as x } from './a.js';
+    `,
+    // Skip the success tests because the imported file does not exist.
+    true
```

Comment:
Not ideal but there is no way around the `CannotFindModuleError` unless the entire test abstraction here is rewritten. Either way, it is more important to test that the rule is active for Source 1 thru 4 than it is to test that the rule is inactive for the library parser.